### ISSUE TITLE
monerod excessive hard drive writes

### ISSUE THREAD

**Issue by undeath** [monero-project/monero#8189]
I noticed that monerod writes a lot of data to my drives, a lot more than what would be expected from keeping the blockchain database up to date.

Within 120h (5d) of running my monerod IO stats (reported by systemd) are: IO: 204.7G read, 407.2G written

Those statistics only include keeping up to date with an already synced blockchain. This does not look right.

**Comment by moneromooo-monero** [comment]
Is this while syncing historical blocks or just keeping up with new blocks after initial sync ?

**Comment by undeath** [comment]
just keeping up with new blocks

**Comment by moneromooo-monero** [comment]
Here's something you can try, to see if it changes the write amount:

<+moneromooo> OK. Do you have a setting suggestion for --db-mode to see if that is the reason for the high usage ?
< hyc> I usually suggest fast:async:1000000
< hyc> but it also won't make much difference if there's not enough RAM to cache more dirty pages
< hyc> if they can run with `/usr/bin/time -v` that will give full stats from getrusage()
< hyc> you'd be looking for the pagefault counts to decrease with a looser sync-mode

So: --db-mode fast:async:1000000

**Comment by undeath** [comment]
Looking much better with that setting. Node has been running for 22h now and current stats are `IO: 33.9G read, 8.4G written`. Also RAM consumption is lower (now 3.7gb, before 5gb).

**Comment by moneromooo-monero** [comment]
Still the same after a similar time from what you originally reported ?

**Comment by undeath** [comment]
Yes, significantly less writes with same amount of reads. After 137h (5.7d) the process is at `IO: 212.7G read, 61.9G written`. RAM consumption went up again  and is now at 6.1gb.

**Comment by XfedeX** [comment]
Shouldn't the default DB config be changed?

**Comment by selsta** [comment]
@XfedeX then we get countless complaints again that the DB corrupts on Windows.. that's why it got changed in the first place.

**Comment by hyc** [comment]
Have considered changing the default back for everything except Windows...

**Comment by selsta** [comment]
Somewhat addressed by #9135 and #8839.
